test_runner/src/main/kotlin/ftl/run/platform/android/ResolveApks.kt

```diff
@@ -2,34 +2,35 @@ package ftl.run.platform.android
 
 import com.google.common.annotations.VisibleForTesting
 import ftl.args.AndroidArgs
+import ftl.args.isSanityRobo
+
 import ftl.run.model.AndroidTestContext
 import ftl.run.model.InstrumentationTestContext
 import ftl.run.model.RoboTestContext
 import ftl.run.exception.FlankGeneralError
+import ftl.run.model.SanityRoboTestContext
 import ftl.util.asFileReference
 
 @VisibleForTesting
 internal fun AndroidArgs.resolveApks(): List<AndroidTestContext> = listOfNotNull(
-    appApk?.let { appApk ->
-        testApk?.let { testApk ->
-            InstrumentationTestContext(
-                app = appApk.asFileReference(),
-                test = testApk.asFileReference()
-            )
-        } ?: roboScript?.let { roboScript ->
-            RoboTestContext(
-                app = appApk.asFileReference(),
-                roboScript = roboScript.asFileReference()
-            )
-        }
-    }
-).plus(
-    elements = additionalAppTestApks.map {
-        InstrumentationTestContext(
-            app = (it.app ?: appApk)
-                ?.asFileReference()
-                ?: throw FlankGeneralError("Cannot create app-test apks pair for instrumentation tests, missing app apk for test ${it.test}"),
-            test = it.test.asFileReference()
-        )
-    }
+    mainApkContext(),
+    *additionalApksContexts()
 )
+
+private fun AndroidArgs.mainApkContext() = appApk?.let { appApk ->
+    when {
+        testApk != null -> InstrumentationTestContext(app = appApk.asFileReference(), test = testApk.asFileReference())
```

Comment:
same as above

Reply:
Here is a bit tricky because I want to take advantage of smart casting